umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/use_constraints.clj

```diff
@@ -93,23 +101,44 @@
                            ;; expression, the first group of the matching string is returned.
                            (if (string? match) match (second match))))]
-      (str/join matches))))
+      (string/join matches))))
+
+(defn parse-ac
+  "Parse the passed in ISO XML access constraints description and value using the
+   passed in xpath and regular expression in this function."
+  [doc constraints-xpath add-xpath reg-ex]
+  (regex-value doc (str constraints-xpath add-xpath "/gco:CharacterString") reg-ex))
+
+;; These are the old and new regex keys to parse on for access constraint values and descriptions.
+(def old-access-value #"(?s)Restriction Flag:(.+)")
+(def new-access-value #"(?s)Access Constraints Value:(.+)")
+(def old-access-desc #"(?s)Restriction Comment:(.+)")
+(def new-access-desc #"(?s)Access Constraints Description:(.+)")
+
+;; These are the xpath parts to where to look for access constraint elements.
+(def use-limitation-xpath "/gmd:useLimitation")
+(def other-constraints-xpath "/gmd:otherConstraints")
 
 (defn parse-access-constraints
   "If both value and Description are nil, return nil.
-  Otherwise, if Description is nil, assoc it with su/not-provided"
+   Otherwise, if Description is nil, assoc it with su/not-provided.
+   The boolean passed to parse-ac is whether or not the parsing is for :Value or :Description."
   [doc constraints-xpath sanitize?]
-  (let [value (regex-value doc (str constraints-xpath
-                                 "/gmd:otherConstraints/gco:CharacterString")
-               #"(?s)Restriction Flag:(.+)")
+  (let [value (or (parse-ac doc constraints-xpath other-constraints-xpath old-access-value)
+                  (parse-ac doc constraints-xpath other-constraints-xpath new-access-value))
         access-constraints-record
         {:Description (su/truncate
-                       (regex-value doc (str constraints-xpath
-                                         "/gmd:useLimitation/gco:CharacterString")
-                         #"(?s)Restriction Comment: (.+)")
+                       (or (parse-ac doc constraints-xpath use-limitation-xpath old-access-desc)
+                           (parse-ac doc constraints-xpath use-limitation-xpath new-access-desc)
+                           (parse-ac doc constraints-xpath other-constraints-xpath old-access-desc)
+                           (parse-ac doc constraints-xpath other-constraints-xpath new-access-desc))
                        su/ACCESSCONSTRAINTS_DESCRIPTION_MAX
                        sanitize?)
          :Value (when value
-                 (Double/parseDouble value))}]
+                  (try (Double/parseDouble (string/trim value))
+                    (catch Exception e
+                      (warn (str "Exception thrown while trying to parse a non parsable ISO access "
+                                 "constraint value number. The document is " doc))
+                      nil)))}]
```

Comment:
`nil` shouldn't be necessary here, warn already returns nil.

Reply:
I have removed the nil.